test/dutchExchange-Tulip.js

```diff
@@ -42,14 +54,188 @@ const setupContracts = async () => {
   } = contracts)
 }
 
-const getBalance = async (acct, token) => ((await dx.balances.call(token.address, acct)).toNumber())
+contract('DutchExchange --> Tulip Flow --> 1 Seller sells 50 ETHER @ 2:1 price --> only 1 BUYER', (accounts) => {
+  const [master, seller1, seller2, buyer1, buyer2] = accounts
+  // const user = seller1
+  // let userTulips
+  let seller1Balance, initialSellVolume 
+  
+  const startBal = {
+    startingETH: 90..toWei(),
+    startingGNO: 1000..toWei(),
+    ethUSDPrice: 60000,
+    sellingAmount: 50..toWei(), // Same as web3.toWei(50, 'ether')
+  }
+  const { startingETH, startingGNO, ethUSDPrice, sellingAmount } = startBal
+
+  before(async () => {
+    // get contracts
+    await setupContracts()
 
-contract('DutchExchange', (accounts) => {
+    /*
+     * SUB TEST 1: Check passed in ACCT has NO balances in DX for token passed in
+     */
+    seller1Balance = await getBalance(seller1, eth)
+    assert.equal(seller1Balance, 0, 'Seller1 should have 0 balance')
+
+    // set up accounts and tokens[contracts]
+    await setupTest(accounts, contracts, startBal)
+
+    /*
+     * SUB TEST 2: Check passed in ACCT has NO balances in DX for token passed in
+     */
+    seller1Balance = await getBalance(seller1, eth)
+    assert.equal(seller1Balance, startingETH, 'Seller1 should have 90 * 10 ** 18')
+
+    /*
+     * SUB TEST 3: assert both eth and gno get approved by DX
+     */
+    // approve ETH
+    await dx.updateApprovalOfToken(eth.address, true, { from: master })
+    // approve GNO
+    await dx.updateApprovalOfToken(gno.address, true, { from: master })
+
+    assert.equal(await dx.approvedTokens.call(eth.address), true, 'ETH is approved by DX')
+    assert.equal(await dx.approvedTokens.call(gno.address), true, 'GNO is approved by DX')
+
+    /*
+     * SUB TEST 4: create new token pair and assert Seller1Balance = 0 after depositing more than Balance
+     */
+    // add tokenPair ETH GNO
+
+    await dx.addTokenPair(
+      eth.address,
+      gno.address,
+      sellingAmount,  // 50 ether - sellVolume for ETH - takes Math.min of amt passed in OR seller balance
+      0,              // buyVolume for GNO
+      2,              // lastClosingPrice NUM
+      1,              // lastClosingPrice DEN
+      { from: seller1 },
+    )
+    seller1Balance = await getBalance(seller1, eth)
+    assert.equal(seller1Balance, (40).toWei(), 'Seller1 should have 40 balance after new Token Pair add')
+  })
+  it('Check sellVolume', async () => {
+    console.log(`
```

Comment:
Can you replace `console.log` with `log` from `utils.js` so that `--silent` works

Reply:
@Velenir done